src/data-source/collection-browser-data-source.ts

```diff
@@ -0,0 +1,1189 @@
+        volume: result.volume?.value,
+        viewCount: result.downloads?.value ?? 0,
+        weeklyViewCount: result.week?.value,
+        loginRequired,
+        contentWarning,
+      });
+    });
+    this.addPage(pageNumber, tiles);
+    const visiblePages = this.host.currentVisiblePageNumbers;
+    const needsReload = visiblePages.includes(pageNumber);
+    if (needsReload) {
+      this.host.refreshVisibleResults();
+    }
+  }
+
+  /**
+   * Returns the mediatype string for the given search result, taking into account
+   * the special `favorited_search` hit type.
+   * @param result The search result to extract a mediatype from
+   */
+  private getMediatype(result: SearchResult): MediaType {
+    /**
+     * hit_type == 'favorited_search' is basically a new hit_type
+     * - we are getting from PPS.
+     * - which gives response for fav- collection
+     * - having favorited items like account/collection/item etc..
+     * - as user can also favorite a search result (a search page)
+     * - so we need to have response (having fav- items and fav- search results)
+     *
+     * if backend hit_type == 'favorited_search'
+     * - let's assume a "search" as new mediatype
+     */
+    if (result?.rawMetadata?.hit_type === 'favorited_search') {
+      return 'search';
+    }
+
+    return result.mediatype?.value ?? 'data';
+  }
+
+  /**
+   * Returns the input string, but removing one set of quotes from all instances of
+   * ""clauses wrapped in two sets of quotes"". This assumes the quotes are already
+   * URL-encoded.
+   *
+   * This should be a temporary measure to address the fact that the __href__ field
+   * sometimes acquires extra quotation marks during query rewriting. Once there is a
+   * full Lucene parser in place that handles quoted queries correctly, this can likely
+   * be removed.
+   */
+  private collapseRepeatedQuotes(str?: string): string | undefined {
+    return str?.replace(/%22%22(?!%22%22)(.+?)%22%22/g, '%22$1%22');
+  }
+
+  /**
+   * Fetches the aggregation buckets for the given prefix filter type.
+   */
+  private async fetchPrefixFilterBuckets(
+    filterType: PrefixFilterType
+  ): Promise<Bucket[]> {
+    const trimmedQuery = this.host.baseQuery?.trim();
+    if (!this.canPerformSearch) return [];
+
+    const filterAggregationKey = prefixFilterAggregationKeys[filterType];
+    const sortParams = this.host.sortParam ? [this.host.sortParam] : [];
+
+    const params: SearchParams = {
+      ...this.pageSpecifierParams,
+      query: trimmedQuery || '',
+      rows: 0,
+      filters: this.filterMap,
+      // Only fetch the firstTitle or firstCreator aggregation
+      aggregations: { simpleParams: [filterAggregationKey] },
+      // Fetch all 26 letter buckets
+      aggregationsSize: 26,
+    };
+    params.uid = await this.requestUID(
+      { ...params, sort: sortParams },
+      'aggregations'
+    );
+
+    const searchResponse = await this.host.searchService?.search(
```

Comment:
Q: the optional chain here denotes that this may not always have a search service - can we explicitly declare that this class needs a searchService to work? it seems like we will always need the service..?

Reply:
Unfortunately our dependency injection model precludes assuming that app services like this one are always present -- in practice, there is a brief limbo period between when the component is created and when the services are injected, which we account for by triggering an update when the service is passed in.

However, this optional chain in particular is likely unnecessary since we can't reach this point in the method if the search service doesn't exist yet. I'll look at cleaning up the logic here a bit.